index.ts

```diff
@@ -170,15 +191,71 @@ async function main() {
   });
   console.log(safeOpHash);
 
-  const signature = await getNearSignature(safeOpHash);
+  // const signature =
+  //   "0xef028109f94aabaf9cd4fb7f2cce311527567341d3439a40fe15e6e01c45249b790e5e5c1f0803f668219f97a95947a982e4f88d8f9abc675adcdb926ed7504d1c";
+  const signature = await getNearSignature(nearAdapter, safeOpHash);
   console.log(
     await sendUserOperation(
-      { ...unsignedUserOp, signature },
+      asERC4337Payload({ ...unsignedUserOp, signature }),
       await contracts.entryPoint.getAddress(),
     ),
   );
 }
 
+interface UserOpData {
+  sender: string;
+  nonce: string;
+  factory?: string | ethers.Addressable;
+  factoryData?: string | ethers.Addressable;
+  callData: string;
+  verificationGasLimit: string;
+  callGasLimit: string;
+  preVerificationGas: string;
+  maxPriorityFeePerGas: string;
+  maxFeePerGas: string;
+  signature: string;
+}
+
+interface SendUserOpPayload {
+  sender: string;
+  nonce: string;
+  initCode: string;
+  callData: string;
+  maxFeePerGas: string;
+  maxPriorityFeePerGas: string;
+  verificationGasLimit: string;
+  callGasLimit: string;
+  preVerificationGas: string;
+  paymasterAndData: string;
+  signature: string;
```

Comment:
Nit: types can be improved.

Reply:
Yes absolutely, however, it would be better to migrate first to `viem` here because the `ethers` types are not very compatible with `viem`.

see above where I had to explicitly convert `hash: string | Uint8Array` into ``0x${string} | Uint8Array`. I suppose its only this hash that is being passed into the adapter so I guess we could improve it here. 

Lets first get an actual transaction through though 🦾